bench/Micro.Benchmark/Benchmarks/HopeEncoder.cs

```diff
@@ -105,6 +105,18 @@ public State(int size)
 
             public Span<byte> EncodingTable => new Span<byte>(_value).Slice(0, _value.Length / 2);
             public Span<byte> DecodingTable => new Span<byte>(_value).Slice(_value.Length / 2);
+
+            public bool CanGrow => false;
+
+            public void Dispose()
+            {
+                throw new NotImplementedException();
```

Comment:
??

Reply:
1. Do we throw from `Dispose`?
2. Given that we don't see it throwing it means that the `Dispose` isn't called. My concern is that we might not dispose other objects implementing `IEncoderState` which do have something to dispose. Please verify that